src/legends.js

```diff
@@ -10,9 +10,13 @@ const legendRegistry = new Map([
 
 export function legend(options = {}) {
   for (const [key, value] of legendRegistry) {
-    const scale = options[key];
-    if (isObject(scale)) { // e.g., ignore {color: "red"}
-      return value(normalizeScale(key, scale), legendOptions(scale, options));
+    const desc = options[key];
+    if (isObject(desc)) { // e.g., ignore {color: "red"}
+      const scale = normalizeScale(key, desc);
+      return Object.assign(
+        value(scale, legendOptions(desc, options)),
+        {scale: exposeScales({[key]: scale})}
```

Comment:
It’s a little more complicated than this now that we have the combined _symbol_ + _color_ legend. That said I’m not sure it’s worth the trouble of implementing this feature.